opal/mca/smsc/xpmem/smsc_xpmem_internal.h

```diff
@@ -23,6 +23,8 @@ typedef struct xpmem_addr xpmem_addr_t;
 
 typedef int64_t xpmem_segid_t;
 typedef int64_t xpmem_apid_t;
+#else
```

Comment:
I don't know that we really need this error.  Thinking about it more, there's no place that this header file would build that we don't have xpmem.h.  I think this is legacy from when the xpmem code was part of the SM btl.  But it's also not harmful.

Reply:
Yeah. It can be removed.

Reply:
@hjelmn just to be sure, do you mean removing this last else-clause? 
I added it so that building this file without either HAVE_XPMEM_H or HAVE_SN_XPMEM_H would produce a more appropriate error, but I'm fine with removing it.

Reply:
I think the idea is this should be caught in the configure step if it can't be built - not a build error.

Reply:
Actually, Nathan and I are saying more than that.  Remove the check for HAVE_XPMEM_H in the xpmem files.  The component will only build if xpmem is available, at which point xpmem.h is also available.

This code was moved from a different component that was always built (even when XPMEM isn't available), so had to be protected with compile bits.  That's changed, so we can simpiify the code.

On the other hand, in terms of getting this patch submitted, I think removing the change to add the else clause is sufficient.  Let's fix the missing define problem (as your patch does) and worry about the cleanups later.

Reply:
@alex--m did you want to clean up - or should we merge and I can do the cleanup in a subsequent pr?

Reply:
We should fix this PR (whether you do it or alex-m does it, I don't care, but we should fix it in this PR).